src/neoxp/Extensions/ExpressRpcFoundStates.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Neo.IO.Json;
+
+namespace NeoExpress
+{
+    public class ExpressRpcFoundStates
+    {
+        public bool Truncated;
+        public (string key, string value)[] Results = new (string key, string value)[0];
+        public (bool hasValue, byte[]? value) FirstProof;
+        public (bool hasValue, byte[]? value) LastProof;
+
+        public static ExpressRpcFoundStates FromJson(JObject json)
+        {
+            return new ExpressRpcFoundStates
+            {
+                Truncated = json["truncated"].AsBoolean(),
+                Results = ((JArray)json["results"])
+                    .Select(j => (
+                        j["key"].AsString(),
+                        j["value"].AsString()
+                    ))
+                    .ToArray(),
+                FirstProof = ProofFromJson(json["firstProof"]),
+                LastProof = ProofFromJson(json["lastProof"]),
+            };
+        }
+
+        private static (bool hasValue, byte[]? value) ProofFromJson(JObject json)
```

Comment:
param should be `JObject? json` if json can be null

Reply:
`json` cannot be `null`?

Reply:
Null has been called the ["billion dollar mistake"](https://www.lucidchart.com/techblog/2015/08/31/the-worst-mistake-of-computer-science/). C# 8 and beyond attempts to undo this mistake by allowing developers to opt-in to declaring if their variables are expected to be null or not. That's what the question mark after the type name means - `JObject? json` means `json` could be `null`, `JObject json` means that `json` should not be null.

Note, the underlying runtime still allows for null variable instances, the nullable checking is done via flow analysis not runtime enforcement. But I like to be consistent and I've opted all managed neo projects that I manage into tracking nullability this way.